src/main/scala/org/constellation/gossip/state/GossipMessagePathTracker.scala

```diff
@@ -0,0 +1,59 @@
+package org.constellation.gossip.state
+
+import cats.effect.Concurrent
+import cats.effect.concurrent.Ref
+import cats.implicits._
+
+class GossipMessagePathTracker[F[_]: Concurrent, A] {
```

Comment:
We only want to keep track of the last layer in the partition map's responses. Not all nodes in the gossip path. This class should also only be checked during snapshot creation and updates by the p2p api when recieving the responses from the last layer in the partition map

Reply:
>We only want to keep track of the last layer in the partition map's responses

It stores information that we sent something to layer 1 and updates itself only when we receive something from the last layer. It does not store information about what happens in between.


So for example for A -> B -> C -> D and A -> E -> F we store there following information:

```
- Sent to B, waiting for response from D
- Sent to E, waiting for response from F
```

And we update that information when we either get response from last layer or it fails somewhere and we just wait too long (so we assume that path failed)

>updates by the p2p api when recieving the responses from the last layer in the partition map

It works like that. We update state only when we send to layer 1 or when we get response from last layer or when we didn't get response from the last layer on time.

>This class should also only be checked during snapshot creation

Sure :+1: For now we just update that class immediately after receiving response from last layer or after some timeout

Reply:
That being said that class and initial proposer are not aware of what happens on path.

It only stores basic information about messages sent to layer 1 (to eventually retry it) and removes any information after getting message from last layer (so cycle is done)